sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -437,101 +474,130 @@ static int _set(gnrc_netif_t *netif, const gnrc_netapi_opt_t *opt)
-                if (netif->lorawan.otaa) {
-                    mlme_request.type = MLME_JOIN;
-                    mlme_request.join.deveui = netif->lorawan.deveui;
-                    mlme_request.join.appeui = netif->lorawan.appeui;
-                    mlme_request.join.appkey = netif->lorawan.appkey;
-                    mlme_request.join.dr = netif->lorawan.datarate;
-                    gnrc_lorawan_mlme_request(&netif->lorawan.mac,
-                                              &mlme_request, &mlme_confirm);
-                }
-                else {
-                    mlme_request.type = MLME_SET;
-                    mlme_request.mib.type = MIB_ACTIVATION_METHOD;
-                    mlme_request.mib.activation = MLME_ACTIVATION_ABP;
-                    gnrc_lorawan_mlme_request(&netif->lorawan.mac,
-                                              &mlme_request, &mlme_confirm);
-                }
+        }
+        netif->lorawan.datarate = *((uint8_t *)opt->data);
+        break;
+    case NETOPT_LORAWAN_TX_PORT:
+        assert(opt->data_len == sizeof(uint8_t));
+        netif->lorawan.port = *((uint8_t *)opt->data);
+        break;
+    case NETOPT_ACK_REQ:
+        assert(opt->data_len == sizeof(netopt_enable_t));
+        netif->lorawan.ack_req = *((netopt_enable_t *)opt->data);
+        break;
+    case NETOPT_ADDRESS_LONG:
+        assert(opt->data_len == LORAMAC_DEVEUI_LEN);
+        _memcpy_reversed(netif->lorawan.deveui, opt->data,
+                         LORAMAC_DEVEUI_LEN);
+        break;
+    case NETOPT_LORAWAN_JOINEUI:
+        assert(opt->data_len == LORAMAC_JOINEUI_LEN);
+        _memcpy_reversed(netif->lorawan.joineui, opt->data,
+                         LORAMAC_JOINEUI_LEN);
+        break;
+    case NETOPT_LORAWAN_NWKKEY:
+        assert(opt->data_len == LORAMAC_NWKKEY_LEN);
+        memcpy(netif->lorawan.nwkkey, opt->data, LORAMAC_NWKKEY_LEN);
+        break;
+#if IS_ACTIVE(CONFIG_GNRC_LORAWAN_1_1)
+    case NETOPT_LORAWAN_APPKEY:
+        assert(opt->data_len == LORAMAC_APPKEY_LEN);
+        memcpy(netif->lorawan.appkey, opt->data, LORAMAC_APPKEY_LEN);
+        break;
+    case NETOPT_LORAWAN_FNWKSINTKEY:
+        assert(opt->data_len == LORAMAC_FNWKSINTKEY_LEN);
+        _memcpy_reversed(netif->lorawan.fnwksintkey, opt->data,
+                         LORAMAC_FNWKSINTKEY_LEN);
+        break;
+    case NETOPT_LORAWAN_SNWKSINTKEY:
+        assert(opt->data_len == LORAMAC_SNWKSINTKEY_LEN);
+        _memcpy_reversed(netif->lorawan.snwksintkey, opt->data,
+                         LORAMAC_SNWKSINTKEY_LEN);
+        break;
+    case NETOPT_LORAWAN_NWKSENCKEY:
+        assert(opt->data_len == LORAMAC_NWKSENCKEY_LEN);
+        _memcpy_reversed(netif->lorawan.nwksenckey, opt->data,
+                         LORAMAC_NWKSKEY_LEN);
+        break;
+#else
+    case NETOPT_LORAWAN_NWKSKEY:
+    case NETOPT_LORAWAN_FNWKSINTKEY:
+    case NETOPT_LORAWAN_SNWKSINTKEY:
+    case NETOPT_LORAWAN_NWKSENCKEY:
```

Comment:
General question: How does a user with LoRaWAN 1.1 firmware connects to a LoRaWAN 1.0.3 NS?